src/main/java/com/cognite/beam/io/transform/yaml/ReadYamlObjectMap.java

```diff
@@ -0,0 +1,136 @@
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.cognite.beam.io.transform.yaml;
+
+import com.google.auto.value.AutoValue;
+import com.google.common.base.Preconditions;
+import org.apache.beam.sdk.io.FileIO;
+import org.apache.beam.sdk.io.fs.EmptyMatchTreatment;
+import org.apache.beam.sdk.options.ValueProvider;
+import org.apache.beam.sdk.transforms.DoFn;
+import org.apache.beam.sdk.transforms.PTransform;
+import org.apache.beam.sdk.transforms.ParDo;
+import org.apache.beam.sdk.values.PBegin;
+import org.apache.beam.sdk.values.PCollection;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.yaml.snakeyaml.DumperOptions;
+import org.yaml.snakeyaml.LoaderOptions;
+import org.yaml.snakeyaml.Yaml;
+import org.yaml.snakeyaml.constructor.BaseConstructor;
+import org.yaml.snakeyaml.constructor.Constructor;
+import org.yaml.snakeyaml.constructor.SafeConstructor;
+import org.yaml.snakeyaml.representer.Representer;
+
+import java.io.ByteArrayInputStream;
+import java.io.InputStream;
+import java.util.HashMap;
+
+/**
+ * Utility transform for parsing a map of key-value pairs from a YAML config file. The map is returned as a
+ * {@code PCollection<T>}. This transform does perform parsing of the values,
+ * so you need to provide .
+ * <p>
+ * Example uses include using YAML files for hosting a map of config properties.
+ */
+@AutoValue
+public abstract class ReadYamlObjectMap<T> extends PTransform<PBegin, PCollection<T>> {
+    protected final Logger LOG = LoggerFactory.getLogger(this.getClass());
+
+    private static <U> Builder<U> builder() {
+        return new AutoValue_ReadYamlObjectMap.Builder<>();
+    }
+
+    /**
+     * Reads from the given file name.
+     *
+     * @param fileName
+     * @return
+     */
+    public static <U> ReadYamlObjectMap<U> from(ValueProvider<String> fileName, Class<U> klass) {
+        Preconditions.checkNotNull(fileName, "File name cannot be null");
+        return ReadYamlObjectMap.<U>builder()
+                .setFilePath(fileName)
+                .setKlass(klass)
+                .build();
+    }
+
+    /**
+     * Reads from the given file name.
+     *
+     * @param fileName
+     * @return
+     */
+    public static <U> ReadYamlObjectMap<U> from(String fileName, Class<U> klass) {
+        return from(ValueProvider.StaticValueProvider.of(fileName), klass);
+    }
+
+    abstract ReadYamlObjectMap.Builder<T> toBuilder();
+
+    abstract ValueProvider<String> getFilePath();
+
+    abstract Class<T> getKlass();
```

Comment:
Are we able to find a better name? "getTargetClass()", "getOutputClass()" or similar?